v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+            parsed_events[current_parsed_event_count].content[sizeof(parsed_events[current_parsed_event_count].content) - 1] = '\0';
+            current_user_data_field += strlen(current_user_data_field) + 1;
+        }
+        POOR_MANS_ASSERT(strcmp((const char*)metadata, "content") == 0);
+        metadata += sizeof("content") + 1;
+
+        if (current_user_data_field != NULL)
+        {
+            (void)strncpy(parsed_events[current_parsed_event_count].file, current_user_data_field, sizeof(parsed_events[current_parsed_event_count].file));
+            parsed_events[current_parsed_event_count].file[sizeof(parsed_events[current_parsed_event_count].file) - 1] = '\0';
+            current_user_data_field += strlen(current_user_data_field) + 1;
+        }
+        POOR_MANS_ASSERT(strcmp((const char*)metadata, "file") == 0);
+        metadata += sizeof("file") + 1;
+
+        if (current_user_data_field != NULL)
+        {
+            (void)strncpy(parsed_events[current_parsed_event_count].func, current_user_data_field, sizeof(parsed_events[current_parsed_event_count].func));
+            parsed_events[current_parsed_event_count].func[sizeof(parsed_events[current_parsed_event_count].func) - 1] = '\0';
+            current_user_data_field += strlen(current_user_data_field) + 1;
+        }
+        POOR_MANS_ASSERT(strcmp((const char*)metadata, "func") == 0);
+        metadata += sizeof("func") + 1;
+
+        parsed_events[current_parsed_event_count].line = *(int32_t*)current_user_data_field;
+        current_user_data_field+= sizeof(int32_t);
+
+        POOR_MANS_ASSERT(strcmp((const char*)metadata, "line") == 0);
+        metadata += sizeof("line") + 1;
+
+        uint32_t property_index = 0;
+
+        // get all other properties
+        while (current_user_data_field - (const char*)pEventRecord->UserData < pEventRecord->UserDataLength)
+        {
+            // get name and type from metadata
+            (void)strncpy(parsed_events[current_parsed_event_count].properties[property_index].property_name, (const char*)metadata, sizeof(parsed_events[current_parsed_event_count].properties[property_index].property_name));
+            parsed_events[current_parsed_event_count].properties[property_index].property_name[sizeof(parsed_events[current_parsed_event_count].properties[property_index].property_name) - 1] = '\0';
+            metadata += strlen(parsed_events[current_parsed_event_count].properties[property_index].property_name) + 1;
+            parsed_events[current_parsed_event_count].properties[property_index].property_type = *metadata;
+            metadata++;
+
+            switch (parsed_events[current_parsed_event_count].properties[property_index].property_type)
+            {
+            default:
+                POOR_MANS_ASSERT(0);
+                break;
+            case _TlgInSTRUCT | _TlgInChain:
+                parsed_events[current_parsed_event_count].properties[property_index].struct_field_count = *(uint8_t*)metadata;
```

Comment:
>uint8_t [](http://example.com/codeflow?start=108&length=7)

this cast doesn't need to lose the const qualifier.

See it as a pattern, so "here and everywhere".

Also, why is it necessary at all when metadata is "const uint8_t*". I know this is contradicting another comment of mine (phrased as a question) from before.

Reply:
Actually made metadata be const char*. Seems to make for less casts.